packages/edit-post/src/components/visual-editor/index.js

```diff
@@ -257,13 +335,36 @@ export default function VisualEditor( { styles } ) {
 						{ themeSupportsLayout &&
 							! themeHasDisabledLayoutStyles &&
 							! isTemplateMode && (
-								<LayoutStyle
-									selector=".edit-post-visual-editor__post-title-wrapper, .block-editor-block-list__layout.is-root-container"
-									layout={ layout }
-									layoutDefinitions={
-										defaultLayout?.definitions
+								<>
+									<LayoutStyle
+										selector=".edit-post-visual-editor__post-title-wrapper, .block-editor-block-list__layout.is-root-container"
+										layout={ fallbackLayout }
+										layoutDefinitions={
+											globalLayoutSettings?.definitions
+										}
+									/>
+									{ postContentLayoutStyles && (
+										<LayoutStyle
+											layout={ postContentLayout }
+											css={ postContentLayoutStyles }
+											layoutDefinitions={
+												globalLayoutSettings?.definitions
+											}
+										/>
+									) }
+									{
+										// For classic themes using theme.json we want a default content width in the editor.
+										! postContentBlock && (
+											<LayoutStyle
+												selector=".block-editor-block-list__layout.is-root-container"
+												layout={ fallbackLayout }
+												layoutDefinitions={
+													globalLayoutSettings?.definitions
+												}
+											/>
+										)
 									}
```

Comment:
Now that the above `<LayoutStyle>` also includes `.block-editor-block-list__layout.is-root-container`, it looks like we no longer need this conditional for classic themes, because it'll already be covered by the one above?

Reply:
Oh yes, well spotted!